Laboratory/Tests/L3-Prelude/Number.cs

```diff
@@ -316,10 +319,8 @@ internal class Number : PreludeFixture
 		[DatapointSource]
 		public (string FunctionExpression, string CallExpression, EleMessageCode ExpectedError)[] FunctionCallErrorCases =
 		{
-			("Num", "(Bool)", EleMessageCode.ConstraintNotSatisfied),
-			("Num", "(Vector3(5, 5, 5))", EleMessageCode.ConstraintNotSatisfied),
-			("Num.mod", "(0, 0)", EleMessageCode.ArgumentOutOfRange),
-			("Num.mod", "(2, 0)", EleMessageCode.ArgumentOutOfRange),
+			("_(a:Num):Num = Num(a)", "(Bool)", EleMessageCode.ConstraintNotSatisfied),
+			("_(a:Num):Num = Num(a)", "(Vector3(5, 5, 5))", EleMessageCode.ConstraintNotSatisfied),
```

Comment:
Same as with Bool, please add a few tests for non-primitive calls to Num.
Also, why remove Num.mod tests?

Reply:
I moved them to the test case above where they are instead tested against NaN, as that's the expected output once we have Nothing, and it's also what C++ (and C#?) does with mod. ArgumentOutOfRange never really made sense for something like this and I think you've mentioned it before.

Reply:
Yeah makes sense I see now.